api/hypershift/v1beta1/nodepool_types.go

```diff
@@ -382,6 +382,45 @@ type NodePoolManagement struct {
 	// +optional
 	// +kubebuilder:default=false
 	AutoRepair bool `json:"autoRepair"`
+
+	// AutoRepairSettings health checks settings for machines in the NodePool when AutoRepair is enabled.
+	//
+	// +optional
+	AutoRepairSettings *AutoRepairSettings `json:"autoRepairSettings,omitempty"`
+}
+
+type AutoRepairSettings struct {
+	// Machines older than this duration without a node will be considered to have
+	// failed and will be remediated.
+	// If not set, this value is defaulted to 20 minutes.
+	// If you wish to disable this feature, set the value explicitly to 0.
+	//
+	// +optional
```

Comment:
Should we default this to in CEL?

Reply:
Why 20 min?
If we are defaulting in the backend, Let's rather default here explicitly.

Reply:
20min is the current value we use.